llvmlibc-samples/src/llvmlibc/baremetal-semihosting/hello.c

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2024, Arm Limited and affiliates.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//    http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+/* Example that uses heap, string and math library */
```

Comment:
I'm not sure you have used the math library, have you? I can only see a call to `abs`, which is defined in `stdlib.h`, not `math.h`.

The problem with the actual math.h functions in embedded LLVM libc is that they won't make a good demo, because you can't printf the floating-point results, because FP printf is left out in the embedded build profile!

I suppose if you really wanted to demonstrate math.h functions, you could do something that has an integer final result, along the lines of `lround(400000 * atan(1))` to get 314159. But it's probably also worth adding a comment explaining _why_ you picked that example, so that when the user inevitably tries tweaking it to printf a floating-point value, they're warned why it won't work.

Reply:
That will teach me for using the list of math functions in cppreference. Was trying to find one that returned an integer.

I've used `atanf` as llvm-libc doesn't define `atan`. This also showed up that I hadn't defined `int *__llvm_libc_errno()` so I've added that to the docs aswell.